### PR TITLE
FIX: use genversion to include SDK version in request headers

### PR DESCRIPTION
**Pull request by markrmullan**
tl;dr, `esbuild` was having a difficult time including the Abound Node SDK when generating a bundle, and it was complaining about the `import.meta` line that gets patched on during the build process. It seems like that added step was just sweeping the original problem I created under the carpet (the original problem being trying to access a `package.json` field directly at runtime).

This PR incorporates the `genversion` library as a dev dependency, and wires it up to the CI/CD pipeline. This eliminates the need for the `import.meta` monkey-patch, which should allow for the SDK to play better with `esbuild`.

As an example,

```console
$ npm run build

> @withabound/node-sdk@1.1.4 build
> sh ./build.sh

The version module version.ts has outdated or unknown content.
Run the following command to fix:
npx genversion --es6 --semi --double src/util/version.ts


$ echo $?
1
```

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -1,8 +1,12 @@
 #!/bin/bash
+set -e
```
Comment (markrmullan) [inline build.sh:2]: Fail `build.sh` if `verify_version.sh` fails

File: package.json [hunk 2]
```diff
@@ -23,7 +23,8 @@
     "compile": "npx tsc --noEmit",
     "lint": "npx xo",
     "prepublishOnly": "npm run build",
-    "test": "npx jest"
+    "test": "npx jest",
+    "verifyVersion": "./verify_version.sh"
```
Comment (markrmullan) [inline package.json:27]: One of the downsides of having GH Actions invoke this from an NPM script rather than just directly from the shell script itself is that the output becomes a little cluttered with NPM output (https://github.com/withabound/abound-node/runs/5965768678?check_suite_focus=true).

Comment (bmealhouse) [reply package.json:27]: I don't like running this through NPM scripts. Can we see what the output looks like by invoking it from the shell directly?

**Comment by bmealhouse** [comment]
@markrmullan, I wasn't able to review in-depth today, but from a high level, I like where this is heading.

File: package.json (hunk 2 above)
Comment (markrmullan) [reply package.json]: Okay cool, agreed, I also like the output when it's just the shell script being invoked: https://github.com/withabound/abound-node/runs/5984972058?check_suite_focus=true

Just wanted to double check for team prefs, especially since `build.sh` is invoked via `npm run build`

**Review by bmealhouse — APPROVED** [review]
lgtm
